pyinfra/api/operations.py

```diff
@@ -128,18 +128,12 @@ def run_host_op(state, host, op_hash):
             try:
                 status = command.execute(state, host, executor_kwargs)
             except Exception as e:  # Custom functions could do anything, so expect anything!
+                _formatted_exc = format_exception(e)
+                _error_msg = "Unexpected error in Python callback: {0}".format(_formatted_exc)
+                _error_msg_styled = click.style(_error_msg, "red")
+                _error_log = "{0}{1}".format(host.print_prefix, _error_msg_styled)
                 logger.warning(traceback.format_exc())
-                logger.error(
```

Comment:
Heh - this is such a huge improvement to readability, the nesting here makes my eyes burn 😆

Reply:
Yeah, it was a bit confusing with a lot of nesting. I don't know if this is the best solution, but at least it's "flat", a bit of an improvement I would say!